packages/material-ui/src/styles/createTypography.js

```diff
@@ -27,27 +25,18 @@ export default function createTypography(palette, typography) {
     // Tell Material-UI what's the font-size on the html element.
     // 16px is the default font-size used by browsers.
     htmlFontSize = 16,
-    suppressDeprecationWarnings = process.env.MUI_SUPPRESS_DEPRECATION_WARNINGS,
-    useNextVariants = false,
+    // eslint-disable-next-line no-underscore-dangle
+    useNextVariants = false || global.__MUI_USE_NEW_TYPOGRAPHY_VARIANTS__,
     // Apply the CSS properties to all the variants.
     allVariants,
     ...other
   } = typeof typography === 'function' ? typography(palette) : typography;
 
   warning(
-    !Object.keys(other).some(variant => typographyMigration.deprecatedVariants.includes(variant)),
-    'Material-UI: You are passing a deprecated variant to ' +
-      `createTypography. ${typographyMigration.migrationGuideMessage}`,
-  );
-
-  warning(
-    useNextVariants ||
-      !Object.keys(other).some(variant => typographyMigration.restyledVariants.includes(variant)),
-    'Material-UI: You are passing a variant to createTypography ' +
-      'that will be restyled in the next major release, without indicating that you ' +
```

Comment:
That warning is now completely removed.

Reply:
Yes, it was. It's not compatible with theme nesting.